Added changes for new python and libraries versions

DmitryBurnaev opened the pull request:
Hello) Thank you very much for your library. It's so helpful to me and I'd like to use it with fresh dependent libraries.
So... In this PR:
- added new python version (3.10)
- bumped libs (including starlette >=0.21.0)
- fixed small code's problems (missing code in a new version of "webargs" library)
- fixed problems in tests and lint's warnings

DmitryBurnaev committed:
Added changes for new python and libraries versions

- added new python version (3.10)
- bumped install required libs
- fixed small code's problems (missing code in a new version of "webargs" library)
- fixed problems in tests and lint's warnings

sloria commented on webargs_starlette/dict2schema.py:
```diff
@@ -0,0 +1,19 @@
+import marshmallow as ma
+
+
+def dict2schema(dct, *, schema_class=ma.Schema) -> typing.Type[ma.Schema]:
```
This shouldn't be necessary, since webargs-starlette relies on marshmallow 3, which already has `Schema.from_dict`.

DmitryBurnaev replied:
done (changed call in `annotations2schema`)

sloria commented on pyproject.toml:
```diff
@@ -1,3 +1,3 @@
 [tool.black]
 line-length = 88
-target-version = ['py36', 'py37', 'py38']
+target-version = ['py36', 'py37', 'py38', 'py310']
```
suggestion (non-blocking): it'd be good to drop support for python 3.6, since it's EOL.

not a blocker to merging, though.

DmitryBurnaev replied:
ok. I've changed version's list (and rechecked via tox)
```
target-version = ['py37', 'py38', 'py39', 'py310', 'py311']
```

sloria commented:
Sorry for the delay in reviewing this! This looks great. If you can just remove the unnecessary dict2schema code, this should be good to go

DmitryBurnaev committed:
Support for new libraries and python's versions

- added new python version (3.11) (just configs)
- added support for new startlette's TestClass (since 0.21.0 it moved from requests to httpx)
- removed redundant code (dict2schema is unused since marshmallow 3.x)

DmitryBurnaev commented:
Hello! I've added some changes. Could you please re-review it?
There were some problems with new version of starlette.  New version of that lib uses `httpx` instead of `requests` and some updates were needed.
So, I've moved part of logic from webtest-asgi into your library (in order to support new test's logic in the starlette). What do you think about it?

DmitryBurnaev commented:
> Hello! I've added some changes. Could you please re-review it? There were some problems with new version of starlette. New version of that lib uses `httpx` instead of `requests` and some updates were needed. So, I've moved part of logic from webtest-asgi into your library (in order to support new test's logic in the starlette). What do you think about it?

There is an another option:)
Instead of copy-pasting, we can merge PR https://github.com/sloria/webtest-asgi/pull/57  and use the new version of `webtest-asgi` in this library (new version can fix problems, which described below)

DmitryBurnaev commented:
> > Hello! I've added some changes. Could you please re-review it? There were some problems with new version of starlette. New version of that lib uses `httpx` instead of `requests` and some updates were needed. So, I've moved part of logic from webtest-asgi into your library (in order to support new test's logic in the starlette). What do you think about it?
> 
> There is an another option:) Instead of copy-pasting, we can merge PR [sloria/webtest-asgi#57](https://github.com/sloria/webtest-asgi/pull/57) and use the new version of `webtest-asgi` in this library (new version can fix problems, which described below)

I'll use the new version of https://github.com/sloria/webtest-asgi and fix this PR.

DmitryBurnaev committed:
Drop support starlette <0.21.0 + bump webtest-asgi version

DmitryBurnaev commented:
I've finished changes with using new version of webtest-asgi. Could you please review this PR too?

DmitryBurnaev commented:
Hello) I hope you are doing well. Could you review my PR again? It would be great to publish a new version of your package to the PyPI repository :) Have a nice day)

sloria committed:
Update changelog

sloria commented:
Sorry for the delay on this again! life's been busy =/

after this merges, i'll cut a release

sloria commented:
released in 2.1.0
